src/plugin.rs

```diff
@@ -0,0 +1,214 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use pyo3::{prelude::*, types::PyType};
+
+use crate::{DecryptError, EncryptError, IdentityError, RecipientError};
+
+#[pyclass(module = "pyrage.plugin", subclass)]
+#[derive(Clone)]
+pub(crate) struct Callbacks;
+
+/// Hack, because the orphan rule would prevent us from deriving a
+/// foreign trait on a foreign object. Instead, define a newtype.
+#[derive(Clone)]
+pub(crate) struct PyCallbacks(Py<Callbacks>);
+
+#[pymethods]
+#[allow(unused_variables)]
+impl Callbacks {
+    const HINT: &'static str =
+        "Subclass `pyrage.plugin.Callbacks` and override all of its methods for this to work.";
+
+    #[new]
+    fn new() -> Self {
+        Self
+    }
+
+    // It doesn't matter what we do here, but it's easy to panic.
+    fn display_message(&self, message: &str) -> PyResult<()> {
+        unimplemented!("{}", Self::HINT)
```

Comment:
Nitpick: let's make these `PyNotImplementedError` instead, since that's more specific and a suitable public API (vs. PyO3's `PanicException`).